lib/route.js

```diff
@@ -0,0 +1,367 @@
+'use strict'
+
+const FindMyWay = require('find-my-way')
+const proxyAddr = require('proxy-addr')
+const Context = require('./context')
+const { buildMiddie, onRunMiddlewares } = require('./middleware')
+const { hookRunner, hookIterator } = require('./hooks')
+const supportedMethods = ['DELETE', 'GET', 'HEAD', 'PATCH', 'POST', 'PUT', 'OPTIONS']
+const validation = require('./validation')
+const buildSchema = validation.build
+const { buildSchemaCompiler } = validation
+const { beforeHandlerWarning } = require('./warnings')
+
+const {
+  kRoutePrefix,
+  kLogLevel,
+  kHooks,
+  kSchemas,
+  kSchemaCompiler,
+  kContentTypeParser,
+  kReply,
+  kRequest,
+  kMiddlewares,
+  kGlobalHooks,
+  kDisableRequestLogging
+} = require('./symbols.js')
+
+function buildRouting (options) {
+  const router = FindMyWay(options.config)
+
+  const schemaCache = new Map()
+  schemaCache.put = schemaCache.set
+
+  let avvio
+  let fourOhFour
+  let trustProxy
+  let requestIdHeader
+  let querystringParser
+  let requestIdLogLabel
+  let logger
+  let hasLogger
+  let setupResponseListeners
+  let throwIfAlreadyStarted
+  let proxyFn
+  let modifyCoreObjects
+  let genReqId
+  let disableRequestLogging
+
+  let closing = false
+
+  return {
+    fill (options, fastifyArgs) {
```

Comment:
What exactly does `fill` mean? I think it would be helpful to use a more descriptive name here.

Reply:
I accept suggestion because I didn't find anything better 😂

The use case is:
+ the router need to be created and used at runtime from other components
+ the router, in turns, needs some pointer to other objects (like avvio, the 404 router itself etc..)

So that method "fill" the missing components that the router needs to have to works properly at runtime.

Maybe is `setup` better?

Reply:
`setup` sounds good to me 🙂